_books/ion-1-1/src/security-considerations.md

```diff
@@ -0,0 +1,156 @@
+
+Ion 1.1 specifies that `parse_ion` must only accept a literal string or literal blob, and that the resulting values are
+always user values (rather than system values). This ensures that the embedded document cannot be affected by any input
+from the containing document, nor can it have any effect on the encoding context of the containing document.
+The `parse_ion` macro uses an Ion reader, so it will be validated just as any other Ion document.
+
+## Data injection via shared modules
+
+Applications are not required to use [shared modules](modules/shared_modules.md). If an application does use shared 
+modules, it should take steps to ensure that shared modules come from a trusted source and use appropriate measures to
+prevent man-in-the-middle and other attacks that can compromise data while it is in transit.
+
+In many cases, even if an application needs to accept Ion payloads from untrusted sources, it is possible to design a 
+solution in which the shared modules are supplied by a trusted source. For example, in a service-oriented-architecture, 
+the server can host shared modules so that the server does not have to trust the client. (However, this assumes that the
+client trusts the server.)
+
+If shared modules must come from an untrusted source, then applications should take steps to ensure that the shared
+modules originate from the same source as the data that uses them, and they can be treated as if they are one composite
+piece of data from that source.
+
+## Arbitrary-sized values
+
+The Ion specification places no limits on the size of Ion values, so an attacker could send a sufficiently large value,
+it could consume enough system resources to disrupt the application reading the value.
+
+Even though the Ion specification does not have limits on the size of values, all real computer systems have finite 
+resources, so all implementations will have limits in practice.
+Ion implementations _MAY_ set limits on the maximum size of any Ion value for any available metric, including (but not 
+limited to) number of bytes, number of codepoints, number of child values, digits of precision, or number of annotations.
+An implementation _MAY_ allow limits to be configurable by an application that uses the Ion implementation.
+Any limits imposed _SHOULD_ be described in the public documentation of an Ion implementation, unless the limits are
+unknown and/or are dependent on the underlying runtime environment.
+
+## Symbol table and macro table inflation
+ 
+An attacker could try to create an input that results in excessively large symbol and macro tables in the Ion reader 
+that could exhaust the memory of the receiving system and lead to a denial of service.
+
+Although Ion 1.1 does not specify a maximum size for symbol tables or macro tables, Ion implementations _MAY_ impose 
+upper bounds on the size of symbol tables, macro tables, module bindings, and any other direct or indirect component of
+the encoding context.
```

Comment:
I'd suggest that an implementation may also allow the user to configure size limits for any kind of value, including user values.